lms/views/dashboard/api/grading.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from datetime import datetime
+
+from marshmallow import Schema, fields
+from pyramid.view import view_config
+from sqlalchemy import select
+
+from lms.models import (
+    ApplicationInstance,
+    Grouping,
+    LMSUser,
+    LTIRegistration,
+)
+from lms.security import Permissions
+from lms.services import LTIAHTTPService
+from lms.services.dashboard import DashboardService
+from lms.services.lti_grading.factory import LTI13GradingService
+from lms.validation._base import JSONPyramidRequestSchema
+
+LOG = logging.getLogger(__name__)
+
+
+class _GradeSchema(Schema):
+    h_userid = fields.Str(required=True)
+    grade = fields.Float(required=True)
+
+
+class AutoGradeSyncSchema(JSONPyramidRequestSchema):
+    grades = fields.List(fields.Nested(_GradeSchema), required=True)
```

Comment:
The endpoint takes the grades from the frontend. I reckon this makes sense because:

- It saves the BE to doing double work of calculating the grades again.
- It guarantees that what you see in the dashboard is what we are going to send to the LMS

Reply:
I was thinking about this as well, and I also think it makes sense.

It would even allow to eventually let instructors adjust the pre-calculated grade before syncing, right from the dashboard, if at some point we want to allow that.